client/v2/tx/factory.go

```diff
@@ -0,0 +1,697 @@
+	SignUnsignedTx()
+	BuildSignedTx()
+	BuildSimulationTx()
+}
+
+// Factory defines a client transaction factory that facilitates generating and
+// signing an application-specific transaction.
+type Factory struct {
+	keybase          keyring.Keyring
+	cdc              codec.BinaryCodec
+	accountRetriever AccountRetriever
+	ac               address.Codec
+	conn             gogogrpc.ClientConn
+	txConfig         TxConfig
+	txParams         TxParameters
+}
+
+// NewFactory returns a factory
+func NewFactory(keybase keyring.Keyring, cdc codec.BinaryCodec, accRetriever AccountRetriever, txConfig TxConfig, ac address.Codec, conn gogogrpc.ClientConn, parameters TxParameters) (Factory, error) {
+	//if clientCtx.Viper == nil {
+	//	clientCtx = clientCtx.WithViper("")
+	//}
+	//
+	//if err := clientCtx.Viper.BindPFlags(flagSet); err != nil {
+	//	return Factory{}, fmt.Errorf("failed to bind flags to viper: %w", err)
+	//}
+	//
+	//var accNum, accSeq uint64
+	//if clientCtx.Offline {
+	//	if flagSet.Changed(flags.FlagAccountNumber) && flagSet.Changed(flags.FlagSequence) {
+	//		accNum = clientCtx.Viper.GetUint64(flags.FlagAccountNumber)
+	//		accSeq = clientCtx.Viper.GetUint64(flags.FlagSequence)
+	//	} else {
+	//		return Factory{}, fmt.Errorf("account-number and sequence must be set in offline mode")
+	//	}
+	//}
+	//
+	//if clientCtx.Offline && clientCtx.GenerateOnly {
+	//	if clientCtx.ChainID != "" {
+	//		return Factory{}, errors.New("chain ID cannot be used when offline and generate-only flags are set")
+	//	}
+	//} else if clientCtx.ChainID == "" {
+	//	return Factory{}, errors.New("chain ID required but not specified")
+	//}
+	//
+	//signMode := flags.ParseSignModeStr(clientCtx.SignModeStr)
+	//memo := clientCtx.Viper.GetString(flags.FlagNote)
+	//timeoutHeight := clientCtx.Viper.GetUint64(flags.FlagTimeoutHeight)
+	//unordered := clientCtx.Viper.GetBool(flags.FlagUnordered)
+	//
+	//gasAdj := clientCtx.Viper.GetFloat64(flags.FlagGasAdjustment)
+	//gasStr := clientCtx.Viper.GetString(flags.FlagGas)
+	//gasSetting, _ := flags.ParseGasSetting(gasStr)
+	//gasPricesStr := clientCtx.Viper.GetString(flags.FlagGasPrices)
+	//
+	//feesStr := clientCtx.Viper.GetString(flags.FlagFees)
+	f := Factory{
+		keybase:          keybase,
+		cdc:              cdc,
+		accountRetriever: accRetriever,
+		ac:               ac,
+		conn:             conn,
+		txConfig:         txConfig,
+		txParams:         parameters,
+	}
+
+	// Properties that need special parsing
+	//f = f.WithFees(feesStr).WithGasPrices(gasPricesStr)
+	return f, nil
+}
+
+// Prepare ensures the account defined by ctx.GetFromAddress() exists and
+// if the account number and/or the account sequence number are zero (not set),
+// they will be queried for and set on the provided Factory.
+// A new Factory with the updated fields will be returned.
+// Note: When in offline mode, the Prepare does nothing and returns the original factory.
+func (f Factory) Prepare() (Factory, error) {
```

Comment:
Just a note here, Prepare can be used to pack all online queries so they're in one place only so when the offline flag is passed the path to building a tx is more clear (vs no flag)